#139: Create Docker Test Setup for notebook tests (#142)
exasol/ai-lab: doc/developer_guide/notebooks.md

# Notebook Files

DSS repository includes some Jupyter notebooks and scripts to add these notebooks to DSS images, e.g. AMI or Docker Images.

Please add or update the notebook files in folder [exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook](../../exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook).

## Notebook Testing

> We are running tests for the notebooks in Docker Edition of the AI Lab for this we are creating Docker test setup 

Reviewer: ```suggestion
We are running tests for the notebooks in Docker Edition of the AI Lab. For this we are creating Docker test setup 
```

> We are running tests for the notebooks in Docker Edition of the AI Lab for this we are creating Docker test setup 

Reviewer 2: ```suggestion
We are running tests for the notebooks in Docker Edition of the AI Lab. For this we are creating Docker test setup 
```

[test_notebooks_in_dss_docker_image.py](test/integration/test_notebooks_in_dss_docker_image.py) which installs test libraries into the AI Lab Docker Image.
It further creates a new test and Docker Container for each notebook test in [test/notebooks](test/notebooks). 

> Notebook tests need to fit the pattern `nbtest_.py`, to prevent pytest running them outside of Docker setup.

Reviewer 2: ```suggestion
Notebook test names need to fit the pattern `nbtest_.py`, to prevent pytest running them outside of Docker setup.
```

> Notebook tests need to fit the pattern `nbtest_.py`, to prevent pytest running them outside of Docker setup.

Reviewer: ```suggestion
Notebook tests need to fit the pattern `nbtest_*.py`, to prevent pytest running them outside of Docker setup.
```

Environment variables with the prefix `NBTEST_` with which you call
[test_notebooks_in_dss_docker_image.py](test/integration/test_notebooks_in_dss_docker_image.py) are forwarded 
into the Docker container and to the notebook test. You can use this to forward secrets to the notebook tests.

> Per default all created containers and images are removed after running the tests regardless of success or failure.

Reviewer 2: ```suggestion
By default all created containers and images are removed after running the tests regardless of success or failure.
```

> However, with the following pytest commandline parameters you can keep them or reuse them for speed up local testing:

Reviewer: ```suggestion
However, with the following pytest commandline parameters you can keep them or reuse them to speed up local testing:
```
